pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -930,15 +851,20 @@ func (c *Cluster) generateEncryptionOpenBlockContainer(resources v1.ResourceRequ
 }
 
 func (c *Cluster) generateVaultGetKEK(osdProps osdProperties) v1.Container {
+	keyName := osdProps.pvc.ClaimName
+	keyPath := encryptionKeyPath()
+
 	return v1.Container{
-		Name:  blockEncryptionKMSGetKEKInitContainer,
-		Image: c.spec.CephVersion.Image,
-		Command: []string{
-			"/bin/bash",
-			"-c",
-			fmt.Sprintf(getKEKFromVaultWithToken, kms.GenerateOSDEncryptionSecretName(osdProps.pvc.ClaimName), encryptionKeyPath()),
+		Name:    blockEncryptionKMSGetKEKInitContainer,
+		Image:   c.rookVersion,
+		Command: []string{"rook"},
+		Args: []string{
+			"secret",
+			"get",
+			keyName,
+			keyPath,
```

Comment:
What if we made a separate binary and image for this? That way, we wouldn't need to pull the whole huge Rook image. I think that binary would probably not change very often, and it could be very small. A `FROM scratch` image even.

Reply:
I've bee thinking about this too. I guess I'm still wondering how we would configure our pod spec to use it, at least which version to pass and how to handle the upgrades. Let's discuss more :)

Reply:
Since the rook image is likely already available locally (for the operator, and osd prepare pods on most nodes), I would hope to just use the same rook image. If efficiency is the only concern, let's not create a new image.

Reply:
I think the Rook image is definitely available locally on single-node installs, but I'm not sure if that's true for larger environments.

Reply:
Still, if efficiency is the only concern, it doesn't warrant creating a new image IMO.